### PR TITLE
Reduce supported python versions to current ones

### PR DESCRIPTION
**Pull request by woutervb**
Only support python 3.8 and python 3.10.
This also opens up the support for modern versions of Flask & Jinja2

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,7 @@
+Version 0.33
+
+  * Drop support for python < 3.8, so only Ubuntu LTS 20.04 & 22.204
```
Comment (maxiberta) [inline CHANGELOG.rst:3]: `s/22.204/22.04/`

**Review by maxiberta — APPROVED** [review]
LGTM, thanks! (check typo)

File: CHANGELOG.rst (hunk 1 above)
Comment (sparkiegeek) [reply CHANGELOG.rst:3]: also it's Ubuntu 20.04 LTS & 22.04 LTS

Comment (sparkiegeek) [inline CHANGELOG.rst:3]:
```suggestion
  * Drop support for Python < 3.8, so only Ubuntu 20.04 LTS and 22.04 LTS
```

**Comment by lofidevops** [comment]
This PR is blocked because it modifies tox.yml, but GitHub still expects the old jobs to run. Someone needs to update actual branch rules as described below.

From https://github.com/orgs/community/discussions/26698 :

> [Configuration of branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule) is completely separate from your workflows. I assume that at some point in the past you’ve marked the previous jobs as required checks (step seven in the documentation, or search for “required status checks”). Once enabled, those jobs are required by name.
>
> You’ve changed job names, so you need to update the branch protection rule: Remove the old jobs, so waiting for jobs that don’t exist any more doesn’t block merging. And ideally add your new job (the non-matrix one) to that gets checked.
>
> ...
>
> So even though I updated the workflow actions I needed to modify the actual branch rules which were still present.

**Pull request merged**